orangecontrib/geo/utils.py

```diff
@@ -42,3 +43,14 @@ def _all_between(vals, min, max):
                     break
 
     return lat_attr, lon_attr
+
+def find_country(data):
+    """Return most likely region identifier"""
+    assert isinstance(data, Table)
+
+    str_attr = [meta for meta in data.domain.metas if isinstance(meta,
+                                                                 StringVariable)]
+    disc_attr = [disc for disc in data.domain.variables if isinstance(disc,
+                                                             DiscreteVariable)]
+
+    return list(chain(str_attr, disc_attr))
```

Comment:
`find_country`, yet the function returns a list of string metas and nominal variables?

It only ever needs the first element further down. Can avoid iterating over all variables with, better:
```py
def guess_region_attr_name(data: Table):
    """Return the name of the first variable that could specify a region name"""
    var = next(chain((var for var in data.domain.metas if var.is_string),
                     (var for var in data.domain.variables if var.is_discrete)),
               None)
    return var.name if var else None
```

Reply:
Anyway, I'd put this very broad looking function inside owgecoding file. That, or inside Orange core. :)

Reply:
Yeah, I didn't know what to do with it. I'll move it to owgeocoding and implement suggestions.

Reply:
What about discrete vars in metas?

Reply:
naming the generators (storing them in variables) before being chained is also not a bad idea

I also sometimes find for loops with return easier to understand than next:

```
string_vars = (var for var in data.domain.metas if var.is_string)
discrete_vars = (var for var in data.domain.variables if var.is_discrete)
for var in chain(string_vars, discrete_vars):
    return var.name
```

as this is subjective, use whatever looks nice to you :)